File: backend/src/main/java/com/hotelmanagement/model/Room.java
package com.hotelmanagement.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

/**
 * Room Entity - Represents a hostel room
 */
@Entity
@Table(name = "rooms")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Room {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "room_number", unique = true, nullable = false, length = 10)
    private String roomNumber;
    
    @Column(name = "room_type", nullable = false)
    private String roomType; // SINGLE, DOUBLE, TRIPLE, DORMITORY
    
    @Column(name = "fee_per_semester", nullable = false, precision = 10, scale = 2)
    private BigDecimal pricePerNight; // Fee per semester
    
    @Column(nullable = false)
    private String status = "AVAILABLE"; // AVAILABLE, OCCUPIED, MAINTENANCE
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(nullable = false)
    private Integer capacity;
    
    @Column(name = "floor_number", nullable = false)
    private Integer floorNumber;
    
    @Column(name = "block_name", nullable = false, length = 50)
    private String blockName; // Block A, Block B, Boys Hostel, Girls Hostel
    
    @Column(length = 500)
    private String amenities; // Bed, Fan, Table, Chair, Cupboard
}

File: node-backend/src/models/Room.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Room = sequelize.define('Room', {
  id: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    autoIncrement: true
  },
  roomNumber: {
    type: DataTypes.STRING(10),
    allowNull: false,
    unique: true,
    field: 'room_number'
  },
  roomType: {
    type: DataTypes.STRING(50),
    allowNull: false,
    field: 'room_type'
  },
  pricePerNight: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    field: 'fee_per_semester'
  },
  status: {
    type: DataTypes.STRING(50),
    allowNull: false,
    defaultValue: 'AVAILABLE'
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  capacity: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  floorNumber: {
    type: DataTypes.INTEGER,
    allowNull: false,
    field: 'floor_number'
  },
  blockName: {
    type: DataTypes.STRING(50),
    allowNull: false,
    field: 'block_name'
  },
  amenities: {
    type: DataTypes.STRING(500),
    allowNull: true
  }
}, {
  tableName: 'rooms',
  timestamps: false
});

module.exports = Room;
